src/libpmemset/sds.c

```diff
@@ -0,0 +1,351 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2021, Intel Corporation */
+
+/*
+ * sds.c -- implementation of common SDS API
+ */
+
+#include <errno.h>
+#include <libpmem2.h>
+
+#include "libpmemset.h"
+#include "libpmem2.h"
+
+#include "alloc.h"
+#include "config.h"
+#include "file.h"
+#include "os_thread.h"
+#include "pmemset_utils.h"
+#include "ravl.h"
+#include "sds.h"
+#include "sys_util.h"
+
+struct pmemset_sds_record {
+	struct pmemset_sds *sds;
+	struct pmemset_source *src;
+	struct pmem2_map *map;
+	size_t refcount;
+};
+
+static struct pmemset_sds_state {
+	struct ravl *rtree;
```

Comment:
hm, isn't there a part tree already that maps to pmem2 maps? can't we generalize that?